dirty_cat/datasets/tests/test_fetching.py

```diff
@@ -1,400 +1,120 @@
-    Though, to avoid the test being too long,
-    we will download a small dataset (<1000 entries).
-
-    Reference:
-    https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/datasets/tests/test_openml.py
+    Test a function that loads data from OpenML.
     """
+    with TemporaryDirectory() as temp_dir_1, TemporaryDirectory() as temp_dir_2:
+        # Convert to path objects
+        temp_dir_1 = Path(temp_dir_1).absolute()
+        temp_dir_2 = Path(temp_dir_2).absolute()
 
-    # Hard-coded information about the test dataset.
-    # Centralizes used information here.
-    test_dataset = {
-        "id": 50,
-        "desc_start": "**Author**",
-        "url": "https://www.openml.org/d/50",
-        "csv_name": "tic-tac-toe.csv",
-        "dataset_rows_count": 958,
-        "dataset_columns_count": 10,
-    }
-
-    test_data_dir = get_test_data_dir()
-
-    try:  # Try-finally block used to remove the test data directory at the end
+        # Fetch without loading into memory
         try:
-            # First, we want to purposefully test ValueError exceptions.
-            with pytest.raises(ValueError):
-                assert fetch_openml_dataset(dataset_id=0, data_directory=test_data_dir)
-                assert fetch_openml_dataset(
-                    dataset_id=2**32, data_directory=test_data_dir
-                )
-
-            # Valid call
-            returned_info = fetch_openml_dataset(
-                dataset_id=test_dataset["id"],
-                data_directory=test_data_dir,
-            )
-
-        except URLError:
-            warnings.warn(
-                "No internet connection or the website is down, test aborted."
-            )
-            # One could try to manually recreate the tree structure
-            # created by ``fetch_openml()``,  and the
-            # ``.gz`` files within in order to finish the test.
-            pytest.skip(
-                "Exception: Skipping this test because we encountered an "
-                "issue probably related to an Internet connection problem. "
+            dataset_wo_load: _fetching.DatasetInfoOnly = fetching_function(
+                load_dataframe=False, directory=temp_dir_1
             )
+        except (ConnectionError,) as e:
+            pytest.skip(f"Got a network error ({e}), skipping.")
             return
-
-        assert returned_info["description"].startswith(test_dataset["desc_start"])
-        assert returned_info["source"] == test_dataset["url"]
-        assert returned_info["path"] == test_data_dir / test_dataset["csv_name"]
-
-        assert returned_info["path"].is_file()
-
-        dataset = pd.read_csv(
-            returned_info["path"], sep=",", quotechar="'", escapechar="\\"
+        else:
+            assert isinstance(dataset_wo_load, _fetching.DatasetInfoOnly)
+
+        # FIXME: An more elegant way of testing whether the dataset is loaded
+        #  in memory would be to monitor it and expect it to stay quite low.
```

Comment:
Right, but that's hard to do reliably, so I wouldn't consider embarking on this too quickly